### PR TITLE
fix(sql): fix error from queries that involve time interval & JIT search

### PR DESCRIPTION
**Pull request by bluestreak01**
Fix "UnsupportedOperationException" from SQLs, such as

```sql
x where k > '1970-01-21T20:00:00' and b = 'BB' limit -2
```

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/IntervalFwdDataFrameCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/AbstractIntervalDataFrameCursor.java|2|2|100.00%|

File: core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java [hunk 1]
```diff
@@ -575,6 +575,47 @@ public void testCursorForLatestByOnSubQueryWithoutRandomAccessSupport() throws E
         );
     }
 
+    @Test
+    public void testJitIntervalFwdCursorBwdSwitch() throws Exception {
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java:579]: nit: might be a good idea to move these new tests to `AsyncFilteredRecordCursorFactoryTest`. `SqlCodeGeneratorTest` contains way too much tests.

**Review by puzpuzpuz — APPROVED** [review]

File: core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java (hunk 1 above)
Comment (bluestreak01) [reply core/src/test/java/io/questdb/griffin/SqlCodeGeneratorTest.java]: done in https://github.com/questdb/questdb/pull/2393/commits/2f6f7d962344da7c7da10004272d1a2f5080b599

**Review by puzpuzpuz — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/IntervalFwdDataFrameCursorFactory.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/AbstractIntervalDataFrameCursor.java|2|2|100.00%|

**Pull request merged**